### ISSUE TITLE
App hanged after MSAppCenter.start

### ISSUE THREAD

**Issue by uOOOO** [microsoft/appcenter-sdk-apple#1997]
### **Description**

App hanged after MSAppCenter.start. But when I added MSCrashes.disableMachExceptionHandler(), the issue's gone.

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Add MSAppCenter.start in AppDelegate
    MSAppCenter.start("secret", withServices: [MSAnalytics.self, MSCrashes.self])
2. Run App
3. App hanged in first screen

### **Details**

1. Which SDK version are you using?
    - 3.0.0
2. Which OS version did you experience the issue on?
    - iOS 12.4.2
3. Which Cocoapods version are you using (run `pod --version`)?
    - 1.8.4
4. What device version did you see this error on?  Were you using an emulator or a physical device?
    - iPhone 6 physical device
5. What language are you using?
    - [ ] Objective C
    - [x] Swift
6. What third party libraries are you using?
  - AlamofireImage (~> 3.5)
  - AppCenter
  - Cleanse
  - DailymotionPlayerSDK
  - IQKeyboardManagerSwift
  - Moya/RxSwift (~> 14.0.0-alpha.1)
  - PMKVObserver (~> 4.1.2)
  - ReactorKit
  - RealmSwift (~> 3.17.3)
  - RxAlamofire (~> 5.0)
  - RxBlocking (~> 5)
  - RxCocoa (~> 5)
  - RxDataSources (~> 4.0)
  - RxSwift (~> 5)
  - RxTest (~> 5)
  - SpotX-SDK
  - SwiftLint
  - SwiftyBeaver
  - SWXMLHash (~> 4.9.0)
7. Please enable verbose logging for your app using `MSAppCenter.setLogLevel(.verbose)` before your call to `MSAppCenter.start(...)` for Swift, or `[MSAppCenter setLogLevel:MSLogLevelVerbose]` before `[MSAppCenter start: ...]` for Objective C and include the logs here:

```
2020-03-09 18:31:34.531145+0700 APP_NAME[416:35610] [AppCenter] VERBOSE: -[MSDelegateForwarder addTraceBlock:]_block_invoke_2/88 Start buffering traces.
2020-03-09 18:31:34.531359+0700 APP_NAME[416:35610] [AppCenter] DEBUG: -[MSDelegateForwarder setEnabledFromPlistForKey:]_block_invoke/278 Delegate forwarder for info.plist key 'AppCenterAppDelegateForwarderEnabled' enabled. It may use swizzling.
2020-03-09 18:31:34.531478+0700 APP_NAME[416:35610] [AppCenter] VERBOSE: +[MSDelegateForwarder flushTraceBuffer]/106 Stop buffering traces, flushed.
2020-03-09 18:31:34.534622+0700 APP_NAME[416:35610] [AppCenter] DEBUG: -[MSSessionContext init]/42 1 session(s) in the history.
2020-03-09 18:31:34.539721+0700 APP_NAME[416:35610] [AppCenter] VERBOSE: -[MSSessionContext setSessionId:]/66 Stored new session with id:(null) and timestamp: 2020-03-09 11:31:34 +0000.
2020-03-09 18:31:34.540568+0700 APP_NAME[416:35610] [AppCenter] DEBUG: -[MSUserIdContext init]/54 1 userId(s) in the history.
2020-03-09 18:31:34.544758+0700 APP_NAME[416:35610] [AppCenter] INFO: -[MSHttpClient networkStateChanged:]/198 Internet connection is up.
2020-03-09 18:31:34.546549+0700 APP_NAME[416:35610] [AppCenter] INFO: -[MSHttpClient networkStateChanged:]/198 Internet connection is up.
2020-03-09 18:31:34.546717+0700 APP_NAME[416:35610] [AppCenter] INFO: -[MSHttpClient networkStateChanged:]/198 Internet connection is up.
2020-03-09 18:31:34.547246+0700 APP_NAME[416:35610] [AppCenter] DEBUG: -[MSDBStorage initWithSchema:version:filename:]_block_invoke/31 SQLite global configuration successfully updated.
2020-03-09 18:31:34.572444+0700 APP_NAME[416:35610] [AppCenter] INFO: -[MSAppCenter configureWithAppSecret:transmissionTargetToken:fromApplication:]/293 App Center SDK configured successfully.
2020-03-09 18:31:34.588708+0700 APP_NAME[416:35610] [AppCenter] VERBOSE: -[MSAppCenter startServices:withAppSecret:transmissionTargetToken:fromApplication:]/319 Start services MSCrashes, MSAnalytics from an application
2020-03-09 18:31:34.589611+0700 APP_NAME[416:35610] [AppCenterCrashes] DEBUG: -[MSCrashes configureCrashReporterWithUncaughtExceptionHandlerEnabled:]/660 EnableUncaughtExceptionHandler is set to YES
2020-03-09 18:31:34.589735+0700 APP_NAME[416:35610] [AppCenterCrashes] VERBOSE: -[MSCrashes configureCrashReporterWithUncaughtExceptionHandlerEnabled:]/670 Enabled Mach exception handler.
[PLCrashReport] MSplcrash_log_writer_init:352: Could not retreive parent process name: Operation not permitted
2020-03-09 18:31:34.608134+0700 APP_NAME[416:35610] [AppCenterCrashes] DEBUG: -[MSCrashes configureCrashReporterWithUncaughtExceptionHandlerEnabled:]/702 Exception handler successfully initialized.
2020-03-09 18:31:34.608892+0700 APP_NAME[416:35610] [AppCenterCrashes] DEBUG: +[MSApplicationForwarder registerForwarding]/188 Application forwarder for info.plist key 'AppCenterApplicationForwarderEnabled' enabled. It may use swizzling.
2020-03-09 18:31:34.622908+0700 APP_NAME[416:35610] [NetworkInfo] Signal strength query returned error: Error Domain=NSPOSIXErrorDomain Code=13 "Permission denied", descriptor: <CTServiceDescriptor 0x28234c8c0, domain=1, instance=1>
2020-03-09 18:31:34.629672+0700 APP_NAME[416:35610] [AppCenter] VERBOSE: -[MSSessionContext clearSessionHistoryAndKeepCurrentSession:]/88 Cleared old sessions.
2020-03-09 18:31:34.630755+0700 APP_NAME[416:35610] [AppCenter] VERBOSE: -[MSUserIdContext clearUserIdHistory]/124 Cleared old userIds while keeping current userId.
2020-03-09 18:31:34.630908+0700 APP_NAME[416:35610] [AppCenterCrashes] INFO: -[MSCrashes applyEnabledState:]/407 Crashes service has been enabled.
2020-03-09 18:31:34.643437+0700 APP_NAME[416:35610] [AppCenterCrashes] VERBOSE: -[MSCrashes startWithChannelGroup:appSecret:transmissionTargetToken:fromApplication:]/452 Started crash service.
2020-03-09 18:31:34.644614+0700 APP_NAME[416:35792] [AppCenter] VERBOSE: -[MSChannelUnitDefault resumeWithIdentifyingObjectSync:]/491 Identifying object <MSAnalytics: 0x281150cf0> removed from pause lane for channel Analytics.
2020-03-09 18:31:34.645364+0700 APP_NAME[416:35792] [AppCenter] DEBUG: -[MSChannelUnitDefault resumeWithIdentifyingObjectSync:]/493 Resume channel Analytics.
2020-03-09 18:31:34.645618+0700 APP_NAME[416:35792] [AppCenter] VERBOSE: -[MSChannelUnitDefault resumeWithIdentifyingObjectSync:]/491 Identifying object <MSAnalytics: 0x281150cf0> removed from pause lane for channel Analytics/one.
2020-03-09 18:31:34.645726+0700 APP_NAME[416:35792] [AppCenter] DEBUG: -[MSChannelUnitDefault resumeWithIdentifyingObjectSync:]/493 Resume channel Analytics/one.
2020-03-09 18:31:34.645732+0700 APP_NAME[416:35610] [AppCenter] VERBOSE: -[MSSessionContext setSessionId:]/66 Stored new session with id:0957FE10-4BDF-4D15-BD05-8742A8E361CB and timestamp: 2020-03-09 11:31:34 +0000.
2020-03-09 18:31:34.645821+0700 APP_NAME[416:35610] [AppCenterAnalytics] INFO: -[MSSessionTracker renewSessionId]/49 New session ID: 0957FE10-4BDF-4D15-BD05-8742A8E361CB
2020-03-09 18:31:34.646884+0700 APP_NAME[416:35610] [AppCenterCrashes] VERBOSE: -[MSCrashes channel:didPrepareLog:internalId:flags:]/535 Storing a log to Crashes Buffer: (sid: 0957FE10-4BDF-4D15-BD05-8742A8E361CB, type: startSession)
2020-03-09 18:31:34.647032+0700 APP_NAME[416:35610] [AppCenterCrashes] VERBOSE: -[MSCrashes channel:didPrepareLog:internalId:flags:]/546 Found an empty buffer position.
2020-03-09 18:31:34.647238+0700 APP_NAME[416:35610] [AppCenterAnalytics] INFO: -[MSAnalytics applyEnabledState:]/156 Analytics service has been enabled.
2020-03-09 18:31:34.647358+0700 APP_NAME[416:35792] [AppCenter] DEBUG: -[MSChannelUnitDefault enqueueItem:flags:]_block_invoke/154 Saving log, type: startSession, flags: 1.
2020-03-09 18:31:34.647407+0700 APP_NAME[416:35610] [AppCenterAnalytics] VERBOSE: -[MSAnalytics startWithChannelGroup:appSecret:transmissionTargetToken:fromApplication:]/107 Started Analytics service.
2020-03-09 18:31:34.648420+0700 APP_NAME[416:35610] [AppCenterCrashes] VERBOSE: -[MSCrashes channel:didPrepareLog:internalId:flags:]/535 Storing a log to Crashes Buffer: (sid: (null), type: startService)
2020-03-09 18:31:34.648722+0700 APP_NAME[416:35610] [AppCenterCrashes] VERBOSE: -[MSCrashes channel:didPrepareLog:internalId:flags:]/558 Remembering index 0 for oldest timestamp 1583753494.647009.
2020-03-09 18:31:34.648962+0700 APP_NAME[416:35610] [AppCenterCrashes] VERBOSE: -[MSCrashes channel:didPrepareLog:internalId:flags:]/546 Found an empty buffer position.
2020-03-09 18:31:34.671455+0700 APP_NAME[416:35792] [AppCenter] VERBOSE: -[MSLogDBStorage saveLog:withGroupId:flags:]_block_invoke/132 Log is stored with id: '36'
2020-03-09 18:31:34.672004+0700 APP_NAME[416:35792] [AppCenterCrashes] VERBOSE: -[MSCrashes channel:didCompleteEnqueueingLog:internalId:]/588 Deleting a log from buffer with id A5E5C307-BF70-48A5-AE88-CB7DD9796E0F
2020-03-09 18:31:34.672425+0700 APP_NAME[416:35792] [AppCenter] DEBUG: -[MSChannelUnitDefault enqueueItem:flags:]_block_invoke/154 Saving log, type: startService, flags: 1.
2020-03-09 18:31:34.691063+0700 APP_NAME[416:35792] [AppCenter] VERBOSE: -[MSLogDBStorage saveLog:withGroupId:flags:]_block_invoke/132 Log is stored with id: '37'
2020-03-09 18:31:34.691564+0700 APP_NAME[416:35792] [AppCenterCrashes] VERBOSE: -[MSCrashes channel:didCompleteEnqueueingLog:internalId:]/588 Deleting a log from buffer with id 4BADF154-2E5C-4894-8632-6FF4DFF9A0C7
2020-03-09 18:31:34.719044+0700 APP_NAME[416:35610] [framework] CUIThemeStore: No theme registered with id=0
2020-03-09 18:31:38.066608+0700 APP_NAME[416:35793] [AppCenter] VERBOSE: -[MSLogDBStorage loadLogsWithGroupId:limit:excludedTargetKeys:completionHandler:]/222 Load log(s) with id(s) '36' as batch Id:E940EF97-F8F1-4090-92EF-FD4DB9B7D84E
2020-03-09 18:31:38.068447+0700 APP_NAME[416:35793] [AppCenter] DEBUG: -[MSChannelUnitDefault sendLogContainer:]/189 Sending 1/1 log, group Id: Analytics, batch Id: E940EF97-F8F1-4090-92EF-FD4DB9B7D84E, session Id: 0957FE10-4BDF-4D15-BD05-8742A8E361CB, payload:
{
  "sid" : "0957FE10-4BDF-4D15-BD05-8742A8E361CB",
  "timestamp" : "2020-03-09T11:31:34.646Z",
  "device" : {
    "screenSize" : "1334x750",
    "appBuild" : "418",
    "osName" : "iOS",
    "timeZoneOffset" : 420,
    "osVersion" : "12.4.2",
    "carrierName" : "KT",
    "locale" : "en_ID",
    "appNamespace" : "com.app.iphone",
    "osBuild" : "16G114",
    "sdkName" : "appcenter.ios",
    "oemName" : "Apple",
    "model" : "iPhone7,2",
    "sdkVersion" : "3.0.0",
    "appVersion" : "3.8.17"
  },
  "type" : "startSession"
}
2020-03-09 18:31:38.069300+0700 APP_NAME[416:35793] [AppCenter] VERBOSE: -[MSAppCenterIngestion willSendHTTPRequestToURL:withHeaders:]/95 URL: https://in.appcenter.ms/logs?api-version=1.0.0
2020-03-09 18:31:38.069530+0700 APP_NAME[416:35793] [AppCenter] VERBOSE: -[MSAppCenterIngestion willSendHTTPRequestToURL:withHeaders:]/96 Headers: Content-Type = application/json, App-Secret = ****************************ff3510fb, Install-ID = 8225B46D-C97A-4613-838A-DA0A17387547
2020-03-09 18:31:38.073334+0700 APP_NAME[416:35793] [AppCenter] VERBOSE: -[MSLogDBStorage loadLogsWithGroupId:limit:excludedTargetKeys:completionHandler:]/222 Load log(s) with id(s) '37' as batch Id:D1F84D32-7D63-4614-A8A4-23F057015AFD
2020-03-09 18:31:38.073676+0700 APP_NAME[416:35793] [AppCenter] DEBUG: -[MSChannelUnitDefault sendLogContainer:]/189 Sending 1/1 log, group Id: AppCenter, batch Id: D1F84D32-7D63-4614-A8A4-23F057015AFD, session Id: (null), payload:
{
  "device" : {
    "screenSize" : "1334x750",
    "appBuild" : "418",
    "osName" : "iOS",
    "timeZoneOffset" : 420,
    "osVersion" : "12.4.2",
    "carrierName" : "KT",
    "locale" : "en_ID",
    "appNamespace" : "com.app.iphone",
    "osBuild" : "16G114",
    "sdkName" : "appcenter.ios",
    "oemName" : "Apple",
    "model" : "iPhone7,2",
    "sdkVersion" : "3.0.0",
    "appVersion" : "3.8.17"
  },
  "timestamp" : "2020-03-09T11:31:34.648Z",
  "type" : "startService",
  "services" : [
    "Crashes",
    "Analytics"
  ]
}
2020-03-09 18:31:38.074777+0700 APP_NAME[416:35793] [AppCenter] VERBOSE: -[MSAppCenterIngestion willSendHTTPRequestToURL:withHeaders:]/95 URL: https://in.appcenter.ms/logs?api-version=1.0.0
2020-03-09 18:31:38.074925+0700 APP_NAME[416:35793] [AppCenter] VERBOSE: -[MSAppCenterIngestion willSendHTTPRequestToURL:withHeaders:]/96 Headers: Content-Type = application/json, App-Secret = ****************************ff3510fb, Install-ID = 8225B46D-C97A-4613-838A-DA0A17387547
2020-03-09 18:32:45.517058+0700 APP_NAME[416:35792] [AppCenter] VERBOSE: -[MSHttpIngestion printResponse:body:error:]/217 HTTP response received with status code: 200, payload:
{
  "status" : "Success",
  "correlationId" : "e3d69837-b340-4745-bedd-6daae90bcfad",
  "validDiagnosticsIds" : [

  ],
  "throttledDiagnosticsIds" : [

  ]
}
2020-03-09 18:32:45.517341+0700 APP_NAME[416:35792] [AppCenter] DEBUG: -[MSChannelUnitDefault sendLogContainer:]_block_invoke/211 Log(s) sent with success, batch Id:E940EF97-F8F1-4090-92EF-FD4DB9B7D84E.
2020-03-09 18:32:45.537072+0700 APP_NAME[416:35792] [AppCenter] VERBOSE: +[MSLogDBStorage deleteLogsFromDBWithColumnValues:columnName:inOpenedDatabase:]/365 Deletion of log(s) by id with value(s) '36' succeeded.
2020-03-09 18:32:48.690054+0700 APP_NAME[416:35795] [AppCenter] VERBOSE: -[MSHttpIngestion printResponse:body:error:]/217 HTTP response received with status code: 200, payload:
{
  "status" : "AllLogsThrottled",
  "correlationId" : "d5ae7311-155f-4cf4-9966-755d8f4e99a2",
  "validDiagnosticsIds" : [

  ],
  "throttledDiagnosticsIds" : [

  ]
}
2020-03-09 18:32:48.690338+0700 APP_NAME[416:35795] [AppCenter] DEBUG: -[MSChannelUnitDefault sendLogContainer:]_block_invoke/211 Log(s) sent with success, batch Id:D1F84D32-7D63-4614-A8A4-23F057015AFD.
2020-03-09 18:32:48.712307+0700 APP_NAME[416:35795] [AppCenter] VERBOSE: +[MSLogDBStorage deleteLogsFromDBWithColumnValues:columnName:inOpenedDatabase:]/365 Deletion of log(s) by id with value(s) '37' succeeded.

```

**Comment by annakocheshkova** [comment]
Hi @uOOOO. I assume this is the log from the run where your app hangs - right? Is there a debugger attached?
Also, is this issue reproduced on another device?

**Comment by annakocheshkova** [comment]
Hey, we haven't heard from you in a while, do you have any more info? Feel free to re-open this issue once you have anything. Thanks.

**Comment by uOOOO** [comment]
Due to COVID-19, I worked from home for a while so I was not able to get the test devices.
I've tested with two devices, iPhone6(12.4.2) and iPad Air2(13.2.3). And the issue occurred on both of them. Debugger was not attached then.
Our app runs with DRM protection library. And it needs a build tool not Xcode's one. As far as I know, the tool manipulates build result.
I think it impacts AppCenter SDK because our app works on simulator without any problem. And the DRM library is not included in the version for simulator.
I'll report this to company provided the DRM library and let you know if they respond to me.
